### ISSUE TITLE
Fail while installing cbindgen

### ISSUE THREAD

**Issue by melkouri** [mozilla/cbindgen#833]
Hello!

I want to install cbindgen using this command line on Ubuntu:

> cargo install --force cbindgen

I get this error:

> Updating crates.io index
warning: spurious network error (2 tries remaining): [6] Couldn't resolve host name (Could not resolve host: crates.io)
warning: spurious network error (1 tries remaining): [6] Couldn't resolve host name (Could not resolve host: crates.io)
error: failed to download from `https://crates.io/api/v1/crates/cbindgen/0.24.3/download`
Caused by:
  [6] Couldn't resolve host name (Could not resolve host: crates.io)

I checked the DNS and it seems to work, I don't know what is the problem

**Comment by ophilli** [comment]
This issue seems more specific to your connection with crates.io than with cbindgen. Are you still having issues?

**Comment by melkouri** [comment]
Yes, I still have the problem, now with a different error:

> Updating crates.io index
> warning: spurious network error (2 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
> warning: spurious network error (1 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
> error: failed to fetch `https://github.com/rust-lang/crates.io-index`
> 
> Caused by:
>   network failure seems to have happened
>   if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
>   https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
> 
> Caused by:
>   failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)

**Comment by ophilli** [comment]
This definitely isn’t a cbindgen issue. Maybe the Rust [discord or zulip](https://www.rust-lang.org/community) servers would be a better place to get help.

In any case since you’re here - can you share more about your development environment? What is returned when you run `rustup show`? What OS are you using to develop? Is this on your laptop, or somewhere else (like maybe a network restricted build host)? You said you checked DNS and it seems fine - how did you check that?

**Comment by melkouri** [comment]
Okay thanks!

I am working on VS Code using wsl Ubuntu terminal. When I run rustup show I get:

> Default host: x86_64-unknown-linux-gnu
> rustup home:  /home/melkouri/.rustup
> 
> stable-x86_64-unknown-linux-gnu (default)
> rustc 1.69.0 (84c898d65 2023-04-16)

I am not working on my personal laptop, this is my company's. I just tried to download it on my personal computer, and it worked. 
I checked my DNS using this [https://www.lifewire.com/check-dns-settings-5215603](https://www.lifewire.com/check-dns-settings-5215603) and then I flushed all DNS cache.
